### PR TITLE
issue #SB-6203 feat: discussion module

### PR DESCRIPTION
**Pull request by anoophm**

### CODE REVIEW

File: demo/plugins/discussion-service/server/routes.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import { RequestValidator } from './RequestValidator';
+
+export class Router implements IRouter {
+
+	init(app: any, manifest: Manifest, auth?: any) {
+		const server: Server = frameworkAPI.getPluginInstance(manifest.id);
+    const validator: RequestValidator = new RequestValidator();
+    app.post('/create/post', validator.validateCreatePostAPI, (req, res) => server.createPost(req, res));
+    app.post('/read/post', validator.validateReadPostAPI, (req, res) => server.getPost(req, res));
+    app.post('/delete/post', validator.validateDeletePostAPI, (req, res) => server.deletePost(req, res));
```
Comment (harishgilimi) [inline demo/plugins/discussion-service/server/routes.ts]: This should be delete request

File: demo/plugins/discussion-service/server/server.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+  public async getPost(req: Request, res: Response) {
+    const searchQuery = _.pick(req.body.request, ["thread_id", "tag"]);
+    await this.cassandra.instance.post.findAsync(searchQuery)
+    .then(data => this.sendSuccess(req,res,data))
+    .catch(error => this.sendError(req,res, {code:"ERR_READ_POST", msg: error}));
+  }
+
+  public async deletePost(req: Request, res: Response) {
+    const searchQuery = _.pick(req.body.request, ["thread_id", "tag"]);
```
Comment (harishgilimi) [inline demo/plugins/discussion-service/server/server.ts]: Should not be a hard delete

File: demo/plugins/discussion-service/server/telemetryHelper.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+class TelemetryHelper {
+  log(req) {
+    const logEventSuccess = {
+      'context': this.getContext(req),
+      'actor': {
+        'id': ACTOR_ID,
+        'type': 'System'
+      },
+      'tags': [],
```
Comment (harishgilimi) [inline demo/plugins/discussion-service/server/telemetryHelper.ts]: tags should not be empty

File: demo/plugins/review-comment/server/manifest.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
+export const manifest = {
+    "id": "@project-sunbird/review-comment",
+    "name": "Form service",
```
Comment (harishgilimi) [inline demo/plugins/review-comment/server/manifest.ts]: name should be changed

File: demo/plugins/review-comment/server/routes.ts [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+import { RequestValidator } from './RequestValidator';
+
+export class Router implements IRouter {
+
+	init(app: any, manifest: Manifest, auth?: any) {
+		const server: Server = frameworkAPI.getPluginInstance(manifest.id);
+    const validator: RequestValidator = new RequestValidator();
+    app.post('/create/comment', validator.validateCreateCommentAPI, (req, res) => server.createComment(req, res));
+    app.post('/read/comment', validator.validateReadCommentAPI, (req, res) => server.getCommentList(req, res));
+    app.post('/delete/comment', validator.validateDeleteCommentAPI, (req, res) => server.deleteComments(req, res));
```
Comment (harishgilimi) [inline demo/plugins/review-comment/server/routes.ts]: This should be a delete request

File: demo/plugins/discussion-service/server/RequestValidator/index.ts [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+        responseCode: "CLIENT_ERROR"
+      }));
+      telemetryHelper.error(req, res, error);
+    } else {
+      next()
+    }
+  }
+
+  validateReadPostAPI(req, res, next){
+    const schema = Joi.object().keys({
```
[the remark this answers is not in the document]
Comment (anoophm) [reply demo/plugins/discussion-service/server/RequestValidator/index.ts]: as discussed we can use this package

**Pull request merged**